src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/InitializeStaticFieldsInline.cs

```diff
@@ -45,39 +46,46 @@ public abstract class InitializeStaticFieldsInlineAnalyzer<TLanguageKindEnum> :
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(CA1810Rule, CA2207Rule);
 
-        protected abstract bool InitialiesStaticField(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken);
-        protected abstract TLanguageKindEnum AssignmentNodeKind { get; }
-
         public override void Initialize(AnalysisContext analysisContext)
         {
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            analysisContext.RegisterCodeBlockStartAction<TLanguageKindEnum>(codeBlockStartContext =>
+            analysisContext.RegisterOperationBlockStartAction(context =>
             {
-                if (!(codeBlockStartContext.OwningSymbol is IMethodSymbol methodSym) ||
-                    !methodSym.IsStatic ||
-                    methodSym.MethodKind != MethodKind.SharedConstructor)
+                if (!(context.OwningSymbol is IMethodSymbol method) ||
+                    !method.IsStatic ||
+                    method.MethodKind != MethodKind.StaticConstructor)
                 {
                     return;
                 }
 
                 var initializesStaticField = false;
```

Comment:
Is it safe to only use a `bool` for a multi-threaded context? I have kept what was used previously.